ui/app/components/ui/searchable-item-list/list-item-search/list-item-search.component.js

```diff
@@ -0,0 +1,100 @@
+import React, { useState, useEffect, useRef } from 'react'
+import PropTypes from 'prop-types'
+import Fuse from 'fuse.js'
+import InputAdornment from '@material-ui/core/InputAdornment'
+import TextField from '../../text-field'
+
+const generateFuseKeys = (keys) => {
+  const weight = 1 / keys.length
+  return keys.map((key) => ({ name: key, weight }))
+}
+
+function usePrevious (value) {
+  const ref = useRef()
+  useEffect(() => {
+    ref.current = value
+  })
+  return ref.current
+}
+
+let fuse
+
+export default function ListItemSearch ({
+  onSearch = null,
+  error = '',
+  listToSearch = [],
+  searchByKeys = [],
+  fuseSearchKeys = null,
+  searchPlaceholderText = '',
+  defaultToAll = false,
+}) {
+  const [searchQuery, setSearchQuery] = useState('')
+
+  const handleSearch = (newSearchQuery) => {
+    setSearchQuery(newSearchQuery)
+    const fuseSearchResult = fuse.search(newSearchQuery)
+    onSearch({
+      searchQuery: newSearchQuery,
+      results: defaultToAll && newSearchQuery === '' ? listToSearch : fuseSearchResult,
+    })
+  }
+
+  const didMountRef = useRef(false)
+  useEffect(() => {
+    if (!didMountRef.current) {
+      didMountRef.current = true
+      fuse = new Fuse(listToSearch, {
+        shouldSort: true,
+        threshold: 0.1,
+        location: 0,
+        distance: 100,
+        maxPatternLength: 32,
+        minMatchCharLength: 1,
+        keys: fuseSearchKeys || generateFuseKeys(searchByKeys),
+      })
+    }
+  }, [ searchByKeys, fuseSearchKeys, listToSearch ])
+
+  const previous = usePrevious({ listToSearch }) || []
```

Comment:
I don't think an _array_ is a very good default for this data structure, which is otherwise always an object.
```suggestion
  const previous = usePrevious({ listToSearch }) || {}
```

Though... why is this an object :thinking:

Reply:
fixed in https://github.com/MetaMask/metamask-extension/pull/8877/commits/349280d1070a3be1c6ef2d8994514dc1f782c941